lib/venue.rb

```diff
@@ -1,2 +1,25 @@
 class Venue
-end
+  attr_reader :name, :capacity, :patrons
+  def initialize(name, capacity, patrons = [])
+    @name = name
+    @capacity = capacity
+    @patrons = patrons
+  end
+  def add_patron(patron)
+    @patrons += [patron]
+  end
+  def yell_at_patrons
+    @patrons.each do |patron|
```

Comment:
Nice start to this. Remember that `.each` will always return the original array so we would need to create an accumulator. As this is written it will run forever (this is called an infinite loop). So you would need to do something like this:

```
def yell_at_patrons
  yelled_at = []
  @patrons.each do |patron|
     patron.upcase
   end
   yelled_at
end
```
since you are doing something (upcasing) you need a new container to add that to.